### PR TITLE
Add proper validation for provider url

### LINKED ISSUE #450: Add proper validation for provider URL
ValidateBasic validates the url to be just non-empty string. Rewrite the url validation to consider a absolute url here: https://github.com/ovrclk/akash/blob/master/x/provider/types/msgs.go#L34-L41

### PR DESCRIPTION
**Pull request by akhilkumarpilli**
Fixes #450

### CODE REVIEW

File: x/provider/types/msgs.go [hunk 1]
```diff
@@ -9,9 +11,9 @@ type MsgCreate Provider
 func (msg MsgCreate) Route() string { return RouterKey }
 func (msg MsgCreate) Type() string  { return "create" }
 func (msg MsgCreate) ValidateBasic() error {
+	_, err := url.Parse(msg.HostURI)
 	switch {
-	case len(msg.HostURI) == 0:
-		// TODO: better uri validation
+	case err != nil:
```
Comment (anilCSE) [inline x/provider/types/msgs.go]: We should check for absolute url too.

File: x/provider/types/msgs.go [hunk 2]
```diff
@@ -32,9 +34,9 @@ type MsgUpdate Provider
 func (msg MsgUpdate) Route() string { return RouterKey }
 func (msg MsgUpdate) Type() string  { return "update" }
 func (msg MsgUpdate) ValidateBasic() error {
+	_, err := url.Parse(msg.HostURI)
 	switch {
-	case len(msg.HostURI) == 0:
-		// TODO: better uri validation
+	case err != nil:
 		return ErrInvalidProviderURI
```
Comment (anilCSE) [inline x/provider/types/msgs.go]: Same as above

File: x/provider/types/errors.go [hunk 3]
```diff
@@ -7,4 +7,6 @@ import (
 var (
 	// ErrInvalidProviderURI register error code for invalid provider uri
 	ErrInvalidProviderURI = sdkerrors.Register(ModuleName, 1, "invalid provider: empty host uri")
```
Comment (anilCSE) [inline x/provider/types/errors.go]:
```suggestion
	ErrInvalidProviderURI = sdkerrors.Register(ModuleName, 1, "invalid provider: invalid host uri")
```

**Comment by gosuri** [comment]
👍

**Pull request merged**